rediscluster/slotrange.go

```diff
@@ -173,13 +174,28 @@ func (c *Cluster) updateMappings(slotRanges []redisclusterutil.SlotsRange) {
 		}
 		for _, shard := range newConfig.shards {
 			sumLatency := uint32(0)
-			for _, addr := range shard.addr {
+			minLatencyID := 0
+			minLatency := uint32(math.MaxUint32)
+
+			for i, addr := range shard.addr {
 				node := newConfig.nodes[addr]
-				sumLatency += atomic.LoadUint32(&node.ping)
+				pingLatency := atomic.LoadUint32(&node.ping)
+				if pingLatency < minLatency {
+					minLatency = pingLatency
+					minLatencyID = i
+				}
+
+				sumLatency += pingLatency
 			}
 			for i, addr := range shard.addr {
 				node := newConfig.nodes[addr]
+
 				weight := sumLatency / atomic.LoadUint32(&node.ping)
+				if c.forceMinLatencyReplica == enabled && i == minLatencyID {
```

Comment:
Shouldn't you use an atomic operation here to retrieve a `c.forceMinLatencyReplica` value?

Reply:
You are right. Fixed